File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        for (int n : nums){
            count[n] = 1 + count[n];
        }

        vector<vector<int>> freq(nums.size() + 1);
        for(const auto& p : count){
            freq[p.second].push_back(p.first);
        }
        vector<int> res;
        int l = 0;
        for(int i = freq.size() - 1; i > 0; i--){
            if(l == k){
                break;
            }
            for(int j = 0; j < freq[i].size(); j++){
                res.push_back(freq[i][j]);
                l++;
                if(l == k){
                    break;
                }
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        
        res = []

        for i in range(len(nums)):
            if i and nums[i] == nums[i - 1]:
                continue
            target = -nums[i]
            l, r = i + 1, len(nums) - 1
            while l < r:
                if nums[l] + nums[r] == target:
                    res.append([nums[i], nums[l], nums[r]])
                    l += 1
                    r -= 1
                    while nums[l] == nums[l - 1] and l < r:
                        l += 1
                if nums[l] + nums[r] < target:
                    l += 1
                    continue
                if nums[l] + nums[r] > target:
                    r -= 1
                    continue
        
        return res




